### PR TITLE
improvements/rpc-proto-payload

### PR DESCRIPTION
**Pull request by ShvaykaD**

### CODE REVIEW

File: application/src/test/java/org/thingsboard/server/transport/TransportSqlTestSuite.java [hunk 1]
```diff
@@ -27,12 +27,12 @@
 @RunWith(ClasspathSuite.class)
 @ClasspathSuite.ClassnameFilters({
         "org.thingsboard.server.transport.*.rpc.sql.*Test",
-        "org.thingsboard.server.transport.*.telemetry.timeseries.sql.*Test",
-        "org.thingsboard.server.transport.*.telemetry.attributes.sql.*Test",
-        "org.thingsboard.server.transport.*.attributes.updates.sql.*Test",
-        "org.thingsboard.server.transport.*.attributes.request.sql.*Test",
-        "org.thingsboard.server.transport.*.claim.sql.*Test",
-        "org.thingsboard.server.transport.*.provision.sql.*Test"
+//        "org.thingsboard.server.transport.*.telemetry.timeseries.sql.*Test",
```
Comment (ashvayka) [inline application/src/test/java/org/thingsboard/server/transport/TransportSqlTestSuite.java:30]: WTF is this?

**Comment by ashvayka** [comment]
Please resolve conflicts and change status to non WIP if you like me to merge this.

File: application/src/test/java/org/thingsboard/server/transport/TransportSqlTestSuite.java (hunk 1 above)
Comment (ShvaykaD) [reply application/src/test/java/org/thingsboard/server/transport/TransportSqlTestSuite.java:30]: this is a typo. I will fix this.

**Comment by ShvaykaD** [comment]
> Please resolve conflicts and change status to non WIP if you like me to merge this.

The pull request is incomplete still. I will let you know once it will be ready.

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java [hunk 2]
```diff
@@ -417,6 +423,8 @@ private void validateProtoSchemas(ProtoTransportPayloadConfiguration protoTransp
                     try {
                         validateTransportProtoSchema(protoTransportPayloadTypeConfiguration.getDeviceAttributesProtoSchema(), ATTRIBUTES_PROTO_SCHEMA);
                         validateTransportProtoSchema(protoTransportPayloadTypeConfiguration.getDeviceTelemetryProtoSchema(), TELEMETRY_PROTO_SCHEMA);
+                        validateTransportProtoSchema(protoTransportPayloadTypeConfiguration.getDeviceRpcRequestProtoSchema(), RPC_REQUEST_PROTO_SCHEMA);
```
Comment (ashvayka) [inline dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java:426]: how do you handle backward compatibility? Is it possible to import the profile that does not have these fields? I am talking about the profile that I have exported via JSON before the upgrade?

File: common/data/src/main/java/org/thingsboard/server/common/data/device/profile/ProtoTransportPayloadConfiguration.java [hunk 3]
```diff
@@ -63,20 +66,60 @@ public TransportPayloadType getTransportPayloadType() {
...
 
+    public Descriptors.Descriptor getRpcResponseDynamicMessageDescriptor(String deviceRpcResponseProtoSchema) {
+        return getDescriptor(deviceRpcResponseProtoSchema, RPC_RESPONSE_PROTO_SCHEMA);
+    }
+
+    public DynamicMessage.Builder getRpcRequestDynamicMessageBuilder(String deviceRpcRequestProtoSchema) {
+        return getDynamicMessageBuilder(deviceRpcRequestProtoSchema, RPC_REQUEST_PROTO_SCHEMA);
+    }
+
+    public String getDeviceRpcResponseProtoSchema() {
```
Comment (ShvaykaD) [inline common/data/src/main/java/org/thingsboard/server/common/data/device/profile/ProtoTransportPayloadConfiguration.java:77]: @ashvayka I guess that this should handle the backward compatibility issue?

File: common/data/src/main/java/org/thingsboard/server/common/data/device/profile/ProtoTransportPayloadConfiguration.java [hunk 4]
```diff
@@ -63,20 +66,60 @@ public TransportPayloadType getTransportPayloadType() {
...
+            return "syntax =\"proto3\";\n" +
+                    "package rpc;\n" +
+                    "\n" +
+                    "message RpcResponseMsg {\n" +
+                    "  string payload = 1;\n" +
+                    "}";
+        }
+    }
+
+    public String getDeviceRpcRequestProtoSchema() {
```
Comment (ShvaykaD) [inline common/data/src/main/java/org/thingsboard/server/common/data/device/profile/ProtoTransportPayloadConfiguration.java:90]: @ashvayka and this.

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java (hunk 2 above)
Comment (ShvaykaD) [reply dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java:426]: I have added comments below. Also, I just tried to import the profile without RPC schemas fields and it's got the default values.

**Pull request merged**